### PR TITLE
Fixes #10566 - allow identifier modifications that do not change interface type

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

File: app/models/nic/interface.rb [hunk 1]
```diff
@@ -54,6 +55,18 @@ def fqdn_was
...
     protected
 
+    def identifier_change
+      return true if self.new_record?
+      old_value = self.identifier_was || ''
+      new_value = self.identifier || ''
+
+      %w(. :).each do |forbidden|
+        if old_value.include?(forbidden) != new_value.include?(forbidden)
+          errors.add(:identifier, "Can't add or remove `#{forbidden}` from identifier")
```
Comment (domcleal) [inline app/models/nic/interface.rb]: i18n extraction

File: app/models/nic/interface.rb [hunk 2]
```diff
@@ -54,6 +55,18 @@ def fqdn_was
 
     protected
 
+    def identifier_change
+      return true if self.new_record?
```
Comment (dLobatog) [inline app/models/nic/interface.rb]: I think you can move up this line to the validation and say 
`validate :identifier_change, :unless => -> { |interface| interface.new_record? }`

File: app/models/nic/interface.rb [hunk 3]
```diff
@@ -54,6 +55,18 @@ def fqdn_was
...
+      old_value = self.identifier_was || ''
+      new_value = self.identifier || ''
+
+      %w(. :).each do |forbidden|
```
Comment (dLobatog) [inline app/models/nic/interface.rb:64]: I'd add a comment to explain why these shouldn't be modified.

**Comment by dLobatog** [comment]
@ares Ready to merge after the two nitpicks in my opinion, thanks!

**Comment by ares** [comment]
@eLobato I've fixed both comments, could you give it another round please?

**Comment by dLobatog** [comment]
Merged as 4fd7aa2e6d6e645e57e7b80543e61f0717daf217, thanks @ares!

**Pull request closed without merging**